src/resolver.js

```diff
@@ -25,10 +25,16 @@ exports.resolve = (binaryBlob, path) => {
   while (parts.length) {
     const key = parts.shift()
     if (node[key] === undefined) {
-      throw new Error(`Object has no property '${key}'`)
+      // Check the named links
+      if (node._namedLinks[key] === undefined) {
```

Comment:
I wonder if, instead of adding a `_namedLinks` property, you could just memoize the link name -> CID mapping in this function if, for example, `parts.length` was longer than 1.

IPFS rarely traverses `DAGNode`s by path due to HAMTs meaning there's no 1-to-1 mapping so I'm not sure we're gaining much from this.

Reply:
`_namedLinks` is also used for `tree()`. I could of course there also traverse the links and extract the name dynamically. But that sounds simpler.